alf/algorithms/algorithm.py

```diff
@@ -313,6 +313,10 @@ def train_state_spec(self):
         """Returns the RNN state spec for train_step()."""
         return self._train_state_spec
 
+    def train_state_2_predict_state(self, state):
+        """Convert RNN state for train_step() to RNN state for predict()"""
```

Comment:
Use a better name: convert_train_state_to_predict_state
And assert:
tf.nest.assert_same_structure(train_state_spec, predict_state_spec)

Reply:
renamed